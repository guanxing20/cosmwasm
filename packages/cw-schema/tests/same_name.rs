#![allow(dead_code)]

mod module1 {
    #[derive(cw_schema::Schemaifier)]
    pub struct Test {
        foo: usize,
    }
}

mod module2 {
    #[derive(cw_schema::Schemaifier)]
    pub struct Test {
        bar: f32,
    }
}

#[derive(cw_schema::Schemaifier)]
struct Combined {
    module1: module1::Test,
    module2: module2::Test,
}

#[test]
fn can_handle_same_name_in_different_modules() {
    let schema = cw_schema::schema_of::<Combined>();
    insta::assert_json_snapshot!(schema);
}
