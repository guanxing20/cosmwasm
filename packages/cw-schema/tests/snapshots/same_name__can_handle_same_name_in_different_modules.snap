---
source: packages/cw-schema/tests/same_name.rs
expression: schema
---
{
  "type": "v1",
  "root": 0,
  "definitions": [
    {
      "name": "same_name_Combined",
      "type": "struct",
      "properties": {
        "module1": {
          "value": 1
        },
        "module2": {
          "value": 3
        }
      }
    },
    {
      "name": "same_name_module1_Test",
      "type": "struct",
      "properties": {
        "foo": {
          "value": 2
        }
      }
    },
    {
      "name": "usize",
      "type": "integer",
      "precision": 64,
      "signed": false
    },
    {
      "name": "same_name_module2_Test",
      "type": "struct",
      "properties": {
        "bar": {
          "value": 4
        }
      }
    },
    {
      "name": "f32",
      "type": "float"
    }
  ]
}
